src/js/components/utils/editModal.tsx

```diff
@@ -3,36 +3,90 @@ import Input from '@salesforce/design-system-react/components/input';
 import Modal from '@salesforce/design-system-react/components/modal';
 import Textarea from '@salesforce/design-system-react/components/textarea';
 import i18n from 'i18next';
-import React, { useCallback, useEffect, useRef } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 
-import { useForm } from '@/components/utils';
+import { useForm, useIsMounted } from '@/components/utils';
 import { Project } from '@/store/projects/reducer';
 
 interface EditModalProps {
   project: Project;
   isOpen: boolean;
   toggleModal: (open: boolean) => void;
+  handleClose: () => void;
 }
-const EditModal = ({ project, isOpen, toggleModal }: EditModalProps) => {
+const EditModal = ({
+  project,
+  isOpen,
+  toggleModal,
+  handleClose,
+}: EditModalProps) => {
+  const [saveBlocked, setSaveBlocked] = useState(true);
+  const isMounted = useIsMounted();
   const form = useRef<HTMLFormElement | null>(null);
   const submitButton = useRef<HTMLButtonElement | null>(null);
 
-  const { handleSubmitFromRef } = useForm({
-    fields: { name: '', description: '' },
+  const handleSuccess = () => {
+    setSaveBlocked(true);
+    handleClose();
+  };
+
+  const handleError = () => {
+    if (isMounted.current) {
+      setSaveBlocked(true);
+    }
+  };
+
+  const defaultName = project.name;
+  const defaultDescription = project.description;
+
+  const {
+    inputs,
+    errors,
+    handleInputChange,
+    setInputs,
+    handleSubmit,
+    resetForm,
+  } = useForm({
+    fields: {
+      name: defaultName,
+      description: defaultDescription,
+    },
+    additionalData: project,
+    onSuccess: handleSuccess,
+    onError: handleError,
+    shouldSubscribeToObject: false,
+    objectType: 'project',
+    update: true,
   });
-  const onSubmit = useCallback((e) => {
-    e.preventDefault();
-    if (form) {
-      const { name, description } = form.current;
-      const data = {
-        ...project,
-        name: name.value,
-        description: description.value,
-      };
-      handleSubmitFromRef(data);
+
+  const defaultNameRef = useRef(defaultName);
+  useEffect(() => {
+    const prevDefaultName = defaultNameRef.current;
+    if (defaultName !== prevDefaultName) {
+      setInputs({ ...inputs, name: defaultName });
+      defaultNameRef.current = defaultName;
+    }
+  }, [defaultName, inputs, setInputs]);
+
+  const defaultDescriptionRef = useRef(defaultDescription);
+  useEffect(() => {
+    const prevDefaultDescription = defaultDescriptionRef.current;
+    if (defaultDescription !== prevDefaultDescription) {
+      // enabled the save button when user starts typing
+      setSaveBlocked(false);
+      setInputs({ ...inputs, description: defaultDescription });
+      defaultDescriptionRef.current = defaultDescription;
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
```

Comment:
@jgerigmeyer I think this is similar to how inputs are updated in `submitReview`.  I expect for the mismatch to happen when you start editing the input, but not working as expected.  Before I look further into why, do you see anything missing here?

Reply:
@duggiemitchell This is really close... I think the issue is that the `if (defaultDescription !== prevDefaultDescription)` block only fires when the _default_ description changes -- meaning when the actual model `project.description` changes, not when the textarea changes. The goal of these `useEffect` sections is to make sure the form loads with the right data _after_ it gets changed.

I'm not sure it's worth worrying about `saveBlocked` at all at this point. It really doesn't hurt to let the user save any time, whether the data is changed or not. So I'd just remove that and get the basic functionality working so we can play with it and see how it feels.